src/wwwroot/controllers/PlanCtrl.js

```diff
@@ -174,6 +184,42 @@
       }      
     }
 
+    function isUpgradeablePlan(limit){
+      //Check constraints to validate that plan is upgradeable
+
+      //Set to true all errors to prevent access if service is not working
+      var errorsPlanUpgrade = {'isDKIMError': true, "isDeliveredError": true, "isDeliveredAndDKIMError": true};
+
+      if(limit==undefined){
+        var isConfigRequired = auth.getLimitsByAccount().then(isUpgradeablePlan).catch($log.error);
```

Comment:
i dont think $log is useful to anyone, if any service fails there is already a ErrorHandler file to catch and show an error in every specific case.

Reply:
I will change this approach to custom events to get existing limit status information when is already loaded from PlanController